core/controllers/reader.py

```diff
@@ -332,7 +332,7 @@ def get(self, exploration_id):
                 start_cursor)
         )
         pretest_question_dicts = [
-            question.to_dict() for question in pretest_questions
+            obj['question'].to_dict() for obj in pretest_questions
```

Comment:
Something's wrong here: we shouldn't be passing objects around, they should be domain objects. What is the conceptual name for the composite object -- and can we make a domain object for that?

Ditto throughout this PR.

Reply:
There is no domain object present currently that can store this, though this is basically a part of the QuestionSkillLink model. Will create a new field in QuestionSkillLink object to store this as well.

Reply:
Left a reply on the last comment from where this dict is received here..